stac_fastapi/tests/sfeos_helpers/test_queryables.py

```diff
@@ -0,0 +1,178 @@
+            return cache
+
+    def test_init(self, mock_db_logic):
+        with patch.dict(
+            os.environ, {"VALIDATE_QUERYABLES": "true", "QUERYABLES_CACHE_TTL": "120"}
+        ):
+            cache = QueryablesCache(mock_db_logic)
+            assert cache.validation_enabled is True
+            assert cache.cache_ttl == 120
+
+    def test_reload_settings(self, queryables_cache):
+        with patch.dict(
+            os.environ, {"VALIDATE_QUERYABLES": "false", "QUERYABLES_CACHE_TTL": "300"}
+        ):
+            queryables_cache.reload_settings()
+            assert queryables_cache.validation_enabled is False
+            assert queryables_cache.cache_ttl == 300
+
+    @pytest.mark.asyncio
+    async def test_get_all_queryables_updates_cache(
+        self, queryables_cache, mock_db_logic
+    ):
+        queryables = await queryables_cache.get_all_queryables()
+        assert queryables == {"prop1", "prop2"}
+        mock_db_logic.get_queryables_mapping.assert_called_once()
+
+    @pytest.mark.asyncio
+    async def test_get_all_queryables_uses_cache(self, queryables_cache, mock_db_logic):
+        await queryables_cache.get_all_queryables()
+        mock_db_logic.get_queryables_mapping.assert_called_once()
+
+        # Should use cache now
+        await queryables_cache.get_all_queryables()
+        mock_db_logic.get_queryables_mapping.assert_called_once()
+
+    @pytest.mark.asyncio
+    async def test_get_all_queryables_refresh_after_ttl(
+        self, queryables_cache, mock_db_logic
+    ):
+        await queryables_cache.get_all_queryables()
+        mock_db_logic.get_queryables_mapping.assert_called_once()
+
+        # Simulate time passing
+        queryables_cache._last_updated = time.time() - queryables_cache.cache_ttl - 1
+
+        await queryables_cache.get_all_queryables()
+        assert mock_db_logic.get_queryables_mapping.call_count == 2
+
+    @pytest.mark.asyncio
+    async def test_get_all_queryables_disabled(self, queryables_cache):
+        queryables_cache.validation_enabled = False
+        queryables = await queryables_cache.get_all_queryables()
+        assert queryables == set()
+
+    @pytest.mark.asyncio
+    async def test_validate_valid_fields(self, queryables_cache):
+        await queryables_cache.validate({"prop1"})
+
+    @pytest.mark.asyncio
+    async def test_validate_invalid_fields(self, queryables_cache):
+        with pytest.raises(HTTPException) as excinfo:
+            await queryables_cache.validate({"invalid_prop"})
+        assert excinfo.value.status_code == 400
+        assert "Invalid query fields: invalid_prop" in str(excinfo.value.detail)
+
+    @pytest.mark.asyncio
+    async def test_validate_disabled(self, queryables_cache):
+        queryables_cache.validation_enabled = False
+        await queryables_cache.validate({"invalid_prop"})
+
+
+class TestGlobalFunctions:
+    @pytest.fixture(autouse=True)
+    def reset_global_cache(self):
```

Comment:
We should be able to remove this fixture after the other changes have been made.